### ISSUE TITLE
Gradient Accumulation

### ISSUE THREAD

**Issue by NKDataConv** [NKDataConv/KIZervus#1]
The batch size for training is currently set to 16. However, bigger batch size might be beneficial for performance. Due to memory restriction, bigger batch sizes lead to memory errors. A solution would be to implement gradient accumulation whereby gradients over multiple batches are calculated and applied as soon as the batch size is reached.
Keras does not support Gradient Accumulation. But there are ways to adapt the optimizer and training procedures, see for example here: https://stackoverflow.com/questions/55268762/how-to-accumulate-gradients-for-large-batch-sizes-in-keras
or here: https://github.com/CyberZHG/keras-gradient-accumulation
